docs/rfc/0696-p2p-node.adoc

```diff
@@ -0,0 +1,318 @@
+"`OPTIONAL`" in this document are to be interpreted as described in <<RFC2119>>
+and <<RFC8174>> when, and only when, they appear in all capitals, as shown here.
+
+CBOR <<RFC8949>> datatype definitions are given using the notation devised in
+CDDL <<RFC8610>>. By convention, `struct`-like datatypes are encoded as CBOR
+maps, where the map key is the zero-based numeric index of the field in
+declaration order.
+
+== Protocols
+
+The main purpose of the node is to operate the core protocol of link currently
+implemented in `librad::net::peer`, which currently consists of holding onto the
+futures du jour and dropping them at opportune moments. Furthermore all
+information in form of synchrounous calls and events obtainable internally via
+the sufficiently space-engineered tincans SHALL be made available.
+
+=== Subroutines
+
+This section expands on the auxilliary routines driven by the node on top of the
+core protocol. Where each SHOULD allow to be disabled through configuration.
+
+==== Public API
+
+For other process to be able to talk to the node, it SHALL keep an API
+subroutine running, which offers introspection into its state. See the dedicated
+section for the surface the node will expose.
+
+==== Discovery
+
+In order to guarantee initial connectivity the protocol needs to be given a set
+of bootstrap nodes, while the main means of discovery is part of the membership
+algorithms. Beyond the initial list passed at start-up as source for discovery
+a streaming variant should be - on opt-in basis - made available. Consult the
+<<API>> section for the internals of how that can be achieved by another process.
+
+==== Announcements
+
+Until superseded by graftier ways, currently the main means of dissemination is
+through gossiping new interesting `Have`s. This is achieved by maintainig a
+persisted view on the list of identities and their current refs. During
+periodic runs a delta is created which serves as the basis of the announcements.
+In turn that list is made known to actively connected peers via the core
+protocols gossip.
+
+==== Requests
+
+The core protocol implicitly tracks projects after replication, and explicitely
+with the `track` operation for remotes in the context of a `Urn`. To enable the
+same approach as for announcements where offline changes to the storage can be
+picked up at any point the node is running and online, this subroutine will
+maintain a persisted view and periodically (potentially provoked by a request
+over the <<API>>) build the delta for new tracked `Urn`s and remotes in their
+context.
+
+Note: A blocking issue for that is
+https://github.com/radicle-dev/radicle-link/issues/141[#141]
+
+==== Tracking
+
+Given the ability to configure a set of urn, peers, both or anything observed
+from connected peers, the node will automatically track and replicate. This
+strategy is currently in use in the seed node and SHALL be preserved.
```

Comment:
Does this mean that the `seed` code is going to be obsoleted?

Reply:
Almost all of it will be, the one thing subsumed by the all-might new daemon is the `Tracking` protocol described in the paragraph above - which in turn will be opt-in and has to be explicitly enabled. The rest of the seed code is pretty obsolete with the new architecture and was largely serving the misguided use-case of serving as an API to power pixels showing peer ids and big project numbers. That part is now covered by https://github.com/radicle-dev/radicle-http-api and therefore will not interfere with the operation or requirements of the seed node itself.

Reply:
Ok. Request for inclusion of `rm -r ./seed` in the tracking issue of this RFC.

Reply:
#722